File: src/main/java/spring/springserver/domain/oauth/OAuthRequest.java
package spring.springserver.domain.oauth;

import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import spring.springserver.domain.member.entity.Role;

public record OAuthRequest (

    @NotBlank
    String username,

    @Email
    @NotBlank
    String email,

    @NotBlank
    String name,

    @Enumerated(EnumType.STRING)
    Role role
) {
}

File: src/main/java/spring/springserver/domain/oauth/data/request/OAuthRequest.kt
package spring.springserver.domain.oauth.data.request

import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import spring.springserver.domain.member.entity.Role

/**
 * @author @gnlandkmg(개발)
 * @author @L98293(코틀린 변환)
 */
data class OAuthRequest(

    @field:NotBlank
    val username: String,

    @field:Email
    @field:NotBlank
    val email: String,

    @field:NotBlank
    val name: String,

    @field:Enumerated(EnumType.STRING)
    val role: Role
)